github/GithubObject.py

```diff
@@ -92,6 +93,26 @@ def remove_unset_items(data: Dict[str, Any]) -> Dict[str, Any]:
 Opt = Union[T, _NotSetType]
 
 
+def is_defined(v: Union[T, _NotSetType]) -> TypeGuard[T]:
+    return not isinstance(v, _NotSetType)
+
+
+def is_undefined(v: Any) -> TypeGuard[_NotSetType]:
+    return isinstance(v, _NotSetType)
+
+
+def is_optional(v, type: Type[T]) -> TypeGuard[Opt[T]]:
+    return isinstance(v, _NotSetType) or isinstance(v, type)
+
+
+def is_optional_list_of_type(v, type: Type[T]) -> TypeGuard[List[T]]:
```

Comment:
```suggestion
def is_optional_list(v, type: Type[T]) -> TypeGuard[List[T]]:
```

Reply:
Shouldn't return type be
```suggestion
def is_optional_list_of_type(v, type: Type[T]) -> TypeGuard[Opt[List[T]]]:
```